vcsclient/github.go

```diff
@@ -288,6 +289,47 @@ func (client *GitHubClient) ListOpenPullRequests(ctx context.Context, owner, rep
 	return mapGitHubPullRequestToPullRequestInfoList(pullRequests)
 }
 
+func (client *GitHubClient) GetPullRequest(ctx context.Context, owner, repository string, pullRequestId int) (PullRequestInfo, error) {
+	ghClient, err := client.buildGithubClient(ctx)
+	if err != nil {
+		return PullRequestInfo{}, err
+	}
+	client.logger.Debug(fetchingPullRequestById, repository)
+	pullRequest, _, err := ghClient.PullRequests.Get(ctx, owner, repository, pullRequestId)
```

Comment:
I suggest checking the response code (200) It is possible that this API does not return errors for some response codes:
https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request--status-codes
In such cases, if 200 is not received and the values of the pullRequest variable are empty, this function may panic.

This comment is applicable to all Git providers, and you can find examples in other functions.